CMakeLists.txt

```diff
@@ -20,6 +20,11 @@ include(CTest)
 
 find_package(Threads)
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  # Options for Visual C++ compiler
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
```

Comment:
Could you please describe in the comment that Zc does?

Reply:
> Failing test:
> [  FAILED  ] RandomTest.GenerateRandomBuffer (0 ms)

Unrelated to my commit though. It passed initially, failed after I merged the latest change.

Reply:
> Could you please describe in the comment that Zc does?

Done. Added a very short summary of what's written here:
https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=vs-2019

Reply:
Nice! Thank you!